contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+### Fast iteration speed for product developement 
+
+As mentioned before, Rails is an opinionated framework that does things the "Rails" way. This includes automating trivial tasks by following certain conventions, which can make the development process faster. For example, in a Rails project that is connected to a database, each class is mapped to a table, as described in the Active Record Pattern, and the table name is just a pluralised version of the class's name - the `User` class becomes the `Users` table, the `Person` class becomes the `People` table, etc. This means that you don't need to spend any time deliberating on the name of the tables, nor do you have to juggle multiple concepts while mapping out the database and application, as Rails will automatically connect the two and spare you from deliberating about the exact name of the table. A similar approach is in seen in the usage of resources, since it also automates much of the process.
+
+The automation of these mundane tasks could make development a more enjoyable (more on that in a bit), and faster process. All of these optimizations can make developing an application between 30-40% faster on Rails, while also making it easier to learn, as you do not need to understand HTTP verbs to make a basic Rails application (while mastery of the topic will certainly help). Furthermore, programmers experienced in Rails usually find it easier to start on existing Rails projects, since the conventions followed are usually the same, giving you the same directory structure and route naming conventions everywhere. 
+  
+### Optimized for Programmer Happiness
+
+One of the primary tenets of Ruby was to make it the "Least Surprising Language", where the language was designed to perform exactly as the developer thought would be the most "expected" function. Rails built on that, and was designed to be optimized for Programmer Happiness. This meant re-imagining many of coding's most standard conventions, as well as adding significant bloat, all for the purpose of making the code look "pretty", and inherently "nice" to write (as subjective as that sounds). 
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array "elements" like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
```

Comment:
`readable, intuitive, and inherently happiness-inducing.` -> `readable and inherently happiness-inducing.`

if you agree with my suggestion to include "intuitive" above.

Reply:
Happiness-inducing sounds a little subjective and too strong in my opinion 😅

Reply:
Hahaha the happiness part is a common Rails rhetoric 😛

Reply:
`readable and easy to work with` might be a better description here, because those are the underlying traits that make Rails happiness-inducing.